src/modules/membership/dal/membership.dal.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { UserRoles } from '@prisma/client';
+import { ErrorDal } from 'src/common/dal/error.dal';
+import { PrismaService } from 'src/modules/prisma/prisma.service';
+
+@Injectable()
+export class MembershipDAL {
+  constructor(
+    private prisma: PrismaService,
+    private errorDAL: ErrorDal,
+  ) {}
+
+  async findMembership(membershipId: number, accountId: number) {
+    const membership = await this.prisma.membership.findFirst({
+      where: {
+        id: membershipId,
+        isConfirmed: true,
+        deletedAt: null,
+        accountId,
+      },
+      select: {
+        id: true,
+        role: {
+          select: {
+            roleName: true,
+          },
+        },
+      },
+    });
+
+    if (!membership) {
+      throw new NotFoundException(
+        'Invalid membership or membership not found.',
+      );
+    }
+
+    return membership;
+  }
+
+  async findNotAdminMembership(accountId: number, userId: number) {
+    try {
+      const notAdminMembership = await this.prisma.membership.findFirst({
```

Comment:
please no business logic in dal

Reply:
This is not business logic, it is error handling